Enhanced configuration dev branch

huse-in opened the pull request:
Adding a Enhanced Signal configuration using Angularjs. Enhanced signal configuration should be listed under Signal Configuration. All required libraries and files are added in ~SPM\Scripts\.  

-- Since there is a lot files added and changes it will be good for you to review this changes.

dloweAvenueConsultants reviewed (commented):
-The migration should not be necessary because it has been added to the seed.
-The interfaces for the repositories should be updated so that other implementations will work.

huse-in commented:
@dloweAvenueConsultants  Can you please give me more information or maybe add comments to files that need some updates.

dloweAvenueConsultants reviewed (commented):
- Any time we are adding calls to the database we need to add the call to interface. So for your change to the SignalsRepository you would need to update ISignalsRepository.
-  Also you defined a class in the repository. This should not be in the repository. Normally this would be in the Business Folder with a folder named Enhanced configuration. 
- I don't see that you are handling versioning in your signal calls.
- You are returning all the signals for your count. This may be slow when you have thousands of signals
- In your change to the detector repository your a selecting all the signals and going through them one by one to find a match. Entity framework will do this for you using linq
and a where clause and it will be much quicker.
- Because you have added the menu item to the configuration you don't need to do a migration to explicitly put that in the database. The next time the database is updated it will
automatically be put in.

huse-in commented:
@dloweAvenueConsultants 

I did updates regarding your last comment. 

- Removed functions from Signal Repository since I don't use them.
- Removed class from the signal repository
- I am handling versions you can see that in ~\ATSPM\SPM\Controllers\EnhancedConfigurationController.cs
- Removed GetAllSignalsForEnhancedConfig function ( didn't use it ) 
- Removed Models from migration

Let me know if I need to update something else. 

Regards,
Husein
